src/components/molecules/popover.tsx

```diff
@@ -0,0 +1,87 @@
+import * as React from "react";
+
+import { FloatingFocusManager, FloatingPortal, useMergeRefs } from "@floating-ui/react";
+
+import { PopoverContext, usePopoverContext } from "@contexts";
+import { usePopover } from "@src/hooks";
+import { PopoverOptions } from "@src/interfaces/components";
+
+export const Popover = ({
+	children,
+	modal = false,
+	...restOptions
+}: {
+	children: React.ReactNode;
+} & PopoverOptions) => {
+	const popover = usePopover({ modal, ...restOptions });
+
+	return <PopoverContext.Provider value={popover}>{children}</PopoverContext.Provider>;
+};
+
+interface PopoverTriggerProps {
+	children: React.ReactNode;
+	asChild?: boolean;
+}
+
+export const PopoverTrigger = React.forwardRef<HTMLElement, React.HTMLProps<HTMLElement> & PopoverTriggerProps>(
+	function PopoverTrigger({ children, ...props }, propRef) {
+		const context = usePopoverContext();
+		const childrenRef = (children as any).ref;
+		const ref = useMergeRefs([context.refs.setReference, propRef, childrenRef]);
+
+		return (
+			<button
+				data-state={context.open ? "open" : "closed"}
+				ref={ref}
+				type="button"
+				{...context.getReferenceProps(props)}
+			>
+				{children}
+			</button>
+		);
+	}
+);
+
+export const PopoverContent = React.forwardRef<HTMLDivElement, React.HTMLProps<HTMLDivElement>>(function PopoverContent(
+	{ style, ...props },
+	propRef
+) {
+	const { context: floatingContext, ...context } = usePopoverContext();
+	const ref = useMergeRefs([context.refs.setFloating, propRef]);
+
+	if (!floatingContext.open) return null;
+
+	return (
+		<FloatingPortal>
+			<FloatingFocusManager context={floatingContext} modal={context.modal}>
+				<div
```

Comment:
accessibility add custom className, for example: add border or something else.

Reply:
Added all here:
https://github.com/autokitteh/web-platform/pull/631/commits/4bf69645bdfbc28f24abc8ced80209c32a5a3540